pkgs/development/coq-modules/equations/default.nix

```diff
@@ -59,7 +59,9 @@ stdenv.mkDerivation rec {
     sha256 = param.sha256;
   };
 
-  buildInputs = with coq.ocamlPackages; [ ocaml camlp5 findlib coq ];
+  buildInputs = with coq.ocamlPackages; [ ocaml camlp5 findlib coq ]
+  ++ stdenv.lib.optional (stdenv.lib.versionAtLeast coq.coq-version "8.12") num
```

Comment:
Can you explain what changed that necessitates this starting with 8.12? I've also observed this failure in aac-tactics (cf. coq-community/aac-tactics#61) but it left me puzzled regarding its origin. Note that this fix is not sufficient, you need the same for Bignums (cf. the failing checks).

Reply:
I don’t know. This seems to be a change between the β and the release.

Reply:
Could that change in Coq be related: <https://github.com/coq/coq/pull/12604>?

Reply:
Yes, this is certainly it. And the timeline matches perfectly.

Reply:
Thanks. I’ve also fixed bignums.

Reply:
Have you seen the discussion about this issue in coq/coq#12604? Do you think we should propagate the dependency on `num` and be done with it? Do you understand why this PR had this effect? Enrico was puzzled by it.

Reply:
IIUC, if `ocamlfind` is told not to link a specific library, it tries to resolve it and fails if the library is not there. Therefore, Coq plug-ins that do not use `num` must depend on them so that `ocamlfind` can find out what is the library that should not be linked.

Let’s hide this funny issue under the carpet and have Coq-8.12 propagate `num`. Will Coq get rid of `num` in 8.13?

Thanks Théo for leading this discussion.

Reply:
Thanks, I think this is a good summary and a good solution.

> Will Coq get rid of `num` in 8.13?

I really can't tell.